gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/DiskStoreImpl.java

```diff
@@ -5168,6 +5171,12 @@ public boolean markIndexRecoveryScheduled() {
     }
   }
 
+  public void resetIndexRecoveryState() {
+    synchronized (this.indexRecoveryState) {
+      this.indexRecoveryState[0] = INDEXRECOVERY_UNINIT;
+    }
+  }
+
```

Comment:
It is an old code which was added by you. It needs few states and that is the reason I guess you would have chosen like this. I think we can leave it like this or else file a low priority ticket to be tackled later.